### ISSUE TITLE
binary io?

### ISSUE THREAD

**Issue by anyahelene** [usethesource/vallang#16]
There's a `IValueBinaryReader` and `IValueBinaryWriter`, but the only implementations are in the `old/` subdirectory and are marked as deprecated. 

Is binary IO still supported?

Is there an easy way to deal with IO these days? Like a neat factory that can produce readers and writers for different formats?

**Comment by PaulKlint** [comment]
Hello Anya, binary IO (with compression!) is better supported than ever :-)

In Rascal, have a look at `ValueIO::writeBinaryValueFile(loc file, value val, bool compression = true);`. It has replaced the old implementations you refer to with a very fast new implementation that also yields very concise data.

If you want to use this new implementation directly from Java, have a look at `/vallang/src/main/java/io/usethesource/vallang/io/binary`

**Comment by DavyLandman** [comment]
to give some context, binary IO has been re-implemented, and they have been made more streaming.

You will want to use `IValueInputStream` and `IValueOutputStream`.

They also automatically compress, and try to keep equal references equal, without the costs of the old binary IO library.

the old code is kept around such that when `IValueInputStream` encounters and old style binary value file, it will switch to that reader. And to test that functionality, we've also kept the writer around. But we renamed the package, to force at least an compile-time error.
